api/app/src/main/kotlin/packit/controllers/PinController.kt

```diff
@@ -16,4 +23,25 @@ class PinController(
     fun getPinnedPackets(): ResponseEntity<List<PacketMetadata>> {
         return ResponseEntity.ok(pinService.findAllPinnedPackets())
     }
+
+    @PreAuthorize("@authz.hasGlobalPacketManagePermission(#root)")
+    @PostMapping()
+    fun pinPacket(
+        @RequestBody @Validated packetPin: PinDto,
+    ): ResponseEntity<PinDto> {
+        val packetId = packetPin.packetId
+        val pin = pinService.createPinByPacketId(packetId)
+
+        return ResponseEntity(PinDto(pin.packetId), HttpStatus.CREATED)
+    }
+
+    @PreAuthorize("@authz.hasGlobalPacketManagePermission(#root)")
```

Comment:
should  be `canManageAllPackets`, :smile: .. as that will allow user.manage to hit as well... we assume this in other places as well... since user.manage can add/remove any permission